### PR TITLE
[CELEBORN-1076] Using text/plain content type for prometheus metrics

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?
Refer https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#basic-info

The http content type is better be `text/plain`.

### Why are the changes needed?

As describe in https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#basic-info.

Advantages
```
Human-readable
Easy to assemble, especially for minimalistic cases (no nesting required)
Readable line by line (with the exception of type hints and docstrings)
```
### Does this PR introduce _any_ user-facing change?
The http content type change.

### How was this patch tested?
image

### CODE REVIEW

File: service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala [hunk 1]
```diff
@@ -61,7 +61,7 @@ class HttpRequestHandler(
       HttpVersion.HTTP_1_1,
       HttpResponseStatus.OK,
       Unpooled.copiedBuffer(response, CharsetUtil.UTF_8))
-    res.headers().set(HttpHeaderNames.CONTENT_TYPE, "text/html; charset=UTF-8")
+    res.headers().set(HttpHeaderNames.CONTENT_TYPE, "text/plain; charset=UTF-8\\n")
```
Comment (pan3793) [inline service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala:64]: why tailing \n is required?

Comment (turboFei) [reply service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala]: removed.

**Comment by turboFei** [comment]
cc @waitinfuture @AngersZhuuuu

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.4.0)/branch-0.3(v0.3.2)

**Pull request closed without merging**